openpype/hosts/houdini/api/lib.py

```diff
@@ -747,3 +751,104 @@ def get_camera_from_container(container):
+        project_settings["houdini"]["general"]["update_houdini_var_context"]
+
+    if not houdini_vars_settings["enabled"]:
+        return houdini_vars_to_update
+
+    houdini_vars = houdini_vars_settings["houdini_vars"]
+
+    # No vars specified - nothing to do
+    if not houdini_vars:
+        return houdini_vars_to_update
+
+    # Get Template data
+    template_data = get_current_context_template_data()
+
+    # Set Houdini Vars
+    for item in houdini_vars:
+        # For consistency reasons we always force all vars to be uppercase
+        item["var"] = item["var"].upper()
+
+        # get and resolve template in value
+        item_value = StringTemplate.format_template(
+            item["value"],
+            template_data
+        )
+
+        if item["var"] == "JOB" and item_value == "":
+            # sync $JOB to $HIP if $JOB is empty
+            item_value = os.environ["HIP"]
+
+        if item["is_directory"]:
+            item_value = item_value.replace("\\", "/")
+
+        current_value = hou.hscript("echo -n `${}`".format(item["var"]))[0]
+
+        # sync both environment variables.
+        # because houdini doesn't do that by default
+        # on opening new files
+        os.environ[item["var"]] = current_value
+
+        if current_value != item_value:
+            houdini_vars_to_update.update(
+                {
+                    item["var"]: (
+                        current_value, item_value, item["is_directory"]
+                    )
+                }
+            )
+
+    return houdini_vars_to_update
+
+
+def update_houdini_vars_context():
+    """Update asset context variables"""
+
+    for var, (_old, new, is_directory) in get_context_var_changes().items():
+        if is_directory:
+            try:
+                os.makedirs(new)
+            except OSError as e:
+                if e.errno != errno.EEXIST:
+                    print(
+                        "  - Failed to create ${} dir. Maybe due to "
+                        "insufficient permissions.".format(var)
+                    )
+
+        hou.hscript("set {}={}".format(var, new))
+        os.environ[var] = new
+        print("  - Updated ${} to {}".format(var, new))
+
+
+def update_houdini_vars_context_dialog():
+    """Show pop-up to update asset context variables"""
+    update_vars = get_context_var_changes()
+    if not update_vars:
+        # Nothing to change
+        return
+
+    message = "\n".join(
+        "${}: {} -> {}".format(var, old or "None", new)
+        for var, (old, new, _is_directory) in update_vars.items()
+    )
+    parent = hou.ui.mainQtWindow()
+    dialog = popup.PopupUpdateKeys(parent=parent)
+    dialog.setModal(True)
+    dialog.setWindowTitle("Houdini scene has outdated asset variables")
+    dialog.setMessage(message)
+    dialog.setButtonText("Fix")
+
+    # on_show is the Fix button clicked callback
+    dialog.on_clicked_state.connect(lambda: update_houdini_vars_context())
```

Comment:
Sorry, this could've just been like this:
```suggestion
    dialog.on_clicked_state.connect(update_houdini_vars_context)
```
Should've written better pseudocode. :)